src/compiler/checker.ts

```diff
@@ -7304,17 +7310,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             // accessible to the current enclosing declaration, or gain access to symbols not accessible to the current
             // enclosing declaration. To keep this chain accurate, insert a fake scope into the chain which makes the
             // function's parameters visible.
-            //
-            // If the declaration is in a JS file, then we don't need to do this at all, as there are no annotations besides
-            // JSDoc, which are always outside the function declaration, so are not in the parameter scope.
-            let cleanup: (() => void) | undefined;
-            if (
-                context.enclosingDeclaration
-                && declaration
-                && declaration !== context.enclosingDeclaration
-                && !isInJSFile(declaration)
-                && (some(expandedParams) || some(typeParameters))
-            ) {
+            let cleanupParams: (() => void) | undefined;

```

Comment:
I guess we're not skipping this in JS anymore? That was a perf optimization, so will have to try and see if webpack regresses...

Reply:
I thought part of the point of this was so that JS declaration emit can also benefit more from types written in source. Without proper scoping it's hard to do any reuse in JS files.

Reply:
Both, my original code actually made scopes a whole lot less, so it was meaningful then. But this fake scope thing has been expanded out far more than I ever expected and you've definitely shown cases where my original assumptions were wrong! Perf doesn't seem affected, thankfully.